corehq/apps/app_manager/views/view_generic.py

```diff
@@ -275,8 +275,7 @@ def _make_name(suffix):
 
     context['latest_commcare_version'] = get_commcare_versions(request.user)[-1]
 
-    if (app and app.doc_type in ('Application', 'LinkedApplication')
-            and has_privilege(request, privileges.COMMCARE_LOGO_UPLOADER)):
+    if app and not is_remote_app(app) and has_privilege(request, privileges.COMMCARE_LOGO_UPLOADER):
```

Comment:
Not really applicable to this PR, but If I went through this correctly, this view_generic could get rid of all of the `if app` checks somewhat easily. https://github.com/dimagi/commcare-hq/blob/40af04629217a861696cb75f36b85bd6826df74e/corehq/apps/app_manager/views/view_generic.py#L171-L173 is what happens when there isn't an app, and the only place where app_id is None is from https://github.com/dimagi/commcare-hq/blob/40af04629217a861696cb75f36b85bd6826df74e/corehq/apps/app_manager/urls.py#L88.

I think if that URL just had a direct redirect to the dashboard, the rest of the function could be simplified by assuming app exists

Reply:
That would be an improvement, yeah. I'm inclined to do it piecemeal, since that code isn't really tested and it's easy to make a mistake.